Adding an IaC product page (#12557)
pulumi/docs: content/product/iac.md

---

> title: "Pulumi IaC: Infrastructure as Code"

Reviewer: should the path be /infrastructure-as-code

and then have /iac redirect to it?

just want to make sure the path is SEO optimized

Author: We don't have an IaC page rn. So let's just change it to 'product/infrastructure-as-code/'. There should be no SEO hit for that.

meta_desc: Infrastructure as Code in any programming language. Enable your team to get code to any cloud productively, securely, and reliably.
layout: iac

heading: Pulumi IaC
subheading: |
    Pulumi is a free, open source infrastructure as code tool, and works best with Pulumi Cloud to
    make managing infrastructure secure, reliable, and hassle-free.

overview:
    title: Infrastructure as Code<br/>in any Programming Language
    description: |
        Build and ship infrastructure faster using languages you know and love. Use Pulumi’s open source SDK to provision infrastructure on any cloud.

key_features_above:
    items:
        - title: "Author in any language, deploy to any cloud"
          sub_title: "Pulumi Infrastructure as Code Engine"
          description:

>             Author infrastructure as code (IaC) using programming languages you know and love – including TypeScript/JavaScript, Python, Go, C#, Java, and YAML. Deploy to 150+ providers like AWS, Azure, Google Cloud, and Kubernetes.

Reviewer: ```suggestion
            Author infrastructure as code (IaC) using programming languages you know and love – including TypeScript/JavaScript, Python, Go, C#, Java, and YAML. Deploy to 170+ providers like AWS, Azure, Google Cloud, and Kubernetes.
```

just to match the other comment I made

                  // Export the cluster's kubeconfig.
                  return new Dictionary<string, object?>
                  {
                    ["kubeconfig"] = cluster.Kubeconfig
                  };
                });
            - title: Main.Java
              language: java
              code: |
                import com.pulumi.Context;
                import com.pulumi.Pulumi;
                import com.pulumi.eks.Cluster;

                public class App {
                    public static void main(String[] args) {
                        Pulumi.run(App::stack);
                    }

                    private static void stack(Context ctx) {
                    final var cluster = new Cluster("eks-cluster");
                    ctx.export("kubeconfig", cluster.kubeconfig());
                  }
                }
            - title: Pulumi.yaml
              language: yaml
              code: |
                resources:
                  eks-cluster:
                    type: eks:Cluster
                outputs:
                  kubeconfig: ${cluster.kubeconfig}
          button:
            text: "Learn more about Pulumi Packages"

>             link: "/product/packages/"

Reviewer: oh wow this is an orphaned page. 

should we be putting them to this page instead? https://www.pulumi.com/docs/using-pulumi/pulumi-packages/

Author: This shouldn't be an orphaned page. https://www.pulumi.com/product/packages/ exits and is being linked to from the Pulumi Overview page rn.

          description: |
            Version, review, test, and deploy infrastructure code through the same tools and processes used for your application code.
          image: "/images/product/pulumi-cicd.png"
          button:
            text: "Learn more about CI/CD Integrations"
            link: "/docs/using-pulumi/continuous-delivery/"
          features:
              - title: Version and review
                description: |
                    Manage infrastructure code in Git and approve changes through pull requests.
              - title: Shift left
                description: |
                    Get rapid feedback on your code with fast [unit tests](/docs/using-pulumi/testing/unit/), and run [integration tests](/docs/using-pulumi/testing/integration/) against ephemeral infrastructure.
              - title: Continuous delivery
                description: |
                    [Integrate your CI/CD provider](/docs/using-pulumi/continuous-delivery/) with Pulumi or use GitOps to [manage Kubernetes clusters](/docs/using-pulumi/continuous-delivery/pulumi-kubernetes-operator/).

stats:
    title: Open source. Enterprise ready.
    description: |

>         Pulumi’s Universal Infrastructure as Code CLI and SDK is an [open-source project](https://github.com/pulumi/) that’s supported

Reviewer: ```suggestion
        Pulumi’s Infrastructure as Code CLI and SDK is an [open-source project](https://github.com/pulumi/) that’s supported
```

>         by an active community. We maintain a [public roadmap](/blog/relaunching-pulumis-public-roadmap/) and welcome feedback and contributions.

Reviewer: ```suggestion
        by an active community. We maintain a [public roadmap](https://github.com/orgs/pulumi/projects/44) and welcome feedback and contributions.
```

We should just link direct to the project

    community:
        number: "10,000s"
        description: of community members
    company:
        number: "1,000s"
        description: of companies
    integration:

>         number: "150+"

Reviewer: I think there is more than this now. i think we should do 170+ since https://www.pulumi.com/registry/ is showing 178

        description: Cloud and service integrations

key_features_below:
    items:
        - title: "The fastest and easiest way to use Pulumi IaC at scale"
          sub_title: "Pulumi Cloud"
          description: |
             A fully-managed service for Pulumi IaC plus so much more. Manage and store infrastructure state & secrets, collaborate within teams, view and search infrastructure, and manage security and compliance using Pulumi Cloud.
          image: "/images/product/pulumi-cloud-dashboard.png"
          button:
            text: "Learn more about Pulumi Cloud"

>             link: "/product/pulumi-service/"

Reviewer: should we just link to pulumi-cloud instead of the redirected URL?

          features:
              - title: Pulumi IaC
                description: |
                    Utilize open-source IaC in TypeScript, Python, Go, C#, Java and YAML. Build and distribute reusable components for 150+ cloud & SaaS providers.
              - title: Automate deployment workflows
                description: |
                    Orchestrate secure deployment workflows through GitHub or an API.
              - title: Search and analytics
                description: |
                    View resources from any cloud in one place. Search for resources across clouds with simple queries and filters.
              - title: Pulumi Automation API
                description: |
                    Build custom deployment and CI/CD workflows that integrate with Pulumi Developer Portal, custom portals, or CLIs.
              - title: Developer portals
                description: |
                    Create internal developer portals to distribute infrastructure templates using Pulumi or the Backstage-plugin.

>               - title: State & secrets

Reviewer: Should we call out ESC in here or something?

                description: |
                    Securely store state with built-in secrets manager or bring your own KMS.
              - title: Identity and access control
                description: |
                    Manage teams with SCIM, SAML SSO, GitHub, GitLab, or Atlassian. Set permissions and access tokens.
              - title: Policy enforcement
                description: |
                    Build policy packs from 150 policies or write your own. Leverage compliance-ready policies for any cloud to increase compliance posture and remediation policies to correct violations.
              - title: Audit logs
                description: |
                    Track and store user actions and change history with option to export logs.

get_started:
    title: Getting started

    get_started:
        title: Get started now
        description: |
            Deploy your first app in just five minutes. Follow our tutorials for AWS, Azure, Google Cloud, Kubernetes, and more.
        cta_text: Get Started

>     migrate:

Reviewer: there is a slight formatting error

![image](https://github.com/user-attachments/assets/d132218e-1d1f-4f14-bbe8-264d47f202ff)

Author: @shughes26 Can you help with this?
